include/rocksdb/public_port.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#pragma once
+
+// Best used after all the headers in case windows #defines give you trouble
+// Minimum subset
+
+#ifndef ROCKSDB_PUBLIC_PORT_H__
+#define ROCKSDB_PUBLIC_PORT_H__
+
+#ifdef OS_WIN
+
+#include <stdint.h>
+
+#undef min
+#undef max
+#undef DeleteFile
+#undef GetCurrentTime
+
+#ifndef strcasecmp
+#define strcasecmp _stricmp
+#endif
+
+#ifndef snprintf
+#define snprintf _snprintf
+#endif
+
+#ifndef ROCKSDB_PRIszt
+#define ROCKSDB_PRIszt "Iu"
+#endif
```

Comment:
Also it's better to avoid strcasecmp too. These are common names and might conflict with users' define (notice users need to include public APIs).